Enable k32l2b

jhqian opened the pull request:
Add board support for frdm_k32l2b3

Add board specific pinmux, doc and dts.
Tested with hello_world, synchronization, philosophers, basic/blinky and basic/button. 

Depends on zephyrproject-rtos/hal_nxp#22

jhqian committed:
dts: nxp: kinetis: add pit yaml

- add yaml for pit

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

jhqian committed:
dts: nxp: k32l2b: add soc dt info

- add devicetree info for k32l2b

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

jhqian committed:
west: update header of hal_nxp

- point to the PR #22 for K32L2B SDK import

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

dbkinder commented on boards/arm/frdm_k32l2b3/doc/index.rst:
```diff
@@ -0,0 +1,230 @@
+.. _frdm_k32l2b3:
+
+NXP FRDM-K32L2B3
+##############
```
Title underlines need to be at least as long as the title above it, so add a couple more ``#``s

dbkinder commented on boards/arm/frdm_k32l2b3/doc/index.rst:
```diff
@@ -0,0 +1,230 @@
+path.
+
+Follow the instructions in :ref:`opensda-jlink-onboard-debug-probe` to program
+the `OpenSDA J-Link FRDM-K32L2B3 Firmware`_.
```
I see you've got a "OpenSDA J-Link Generic Firmware" link defined at the bottom of the doc, so this should probably refer to that?

dbkinder commented on boards/arm/frdm_k32l2b3/doc/index.rst:
```diff
@@ -0,0 +1,230 @@
+examples to allow rapid development, plus configuration tools and an optional
+free IDE.
+
+.. image:: ./frdm_k32l2b3.jpg
```
the filename uploaded has a dash in it: ``frdm-k32l2b3.jpg`` so rename the image filename or fix this name to match

jhqian replied:
Thank you, @dbkinder . just updated the doc accordingly.

dbkinder reviewed (approved):
+1 for docs, thanks.

MaureenHelm commented on modules/Kconfig.mcux:
```diff
@@ -6,7 +6,7 @@
 	bool
 	select HAS_CMSIS_CORE
-	depends on SOC_FAMILY_KINETIS || SOC_FAMILY_IMX || SOC_FAMILY_LPC
+	depends on SOC_FAMILY_KINETIS || SOC_FAMILY_IMX || SOC_FAMILY_LPC || SOC_FAMILY_K32L2
```
This should be a new series under the Kinetis family, not a new family.

MaureenHelm commented on boards/arm/frdm_k32l2b3/doc/index.rst:
```diff
@@ -0,0 +1,230 @@
+A debug probe is used for both flashing and debugging the board. This board is
+configured by default to use the :ref:`opensda-daplink-onboard-debug-probe`.
+
+Early versions of this board have an outdated version of the OpenSDA bootloader
```
Is this true for a new board?

jhqian replied:
removed

MaureenHelm commented on boards/arm/frdm_k32l2b3/doc/index.rst:
```diff
@@ -0,0 +1,230 @@
+------------------------------------------------------------------
+
+Install the :ref:`pyocd-debug-host-tools` and make sure they are in your search
+path.
```
pyocd doesn't support this soc yet, does it?

jhqian replied:
remove that statement for now, in fact, I'm working on a PR

hakehuang commented:
@jhqian , I find there is a problem with samples/subsys/usb/cdc_acm with the -DOVERLAY_CONFIG=overlay-composite-cdc-msc.conf, it reports 
```

[ 93%] Built target drivers__serial

Scanning dependencies of target zephyr_prebuilt

[ 94%] Building C object zephyr/CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj

[ 95%] Linking C executable zephyr_prebuilt.elf

Memory region         Used Size  Region Size  %age Used

           FLASH:       54804 B       256 KB     20.91%

            SRAM:       32996 B        32 KB    100.70%

        IDT_LIST:          72 B         2 KB      3.52/opt/zephyr-sdk/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/8.3.0/../../../../arm-zephyr-eabi/bin/ld: zephyr_prebuilt.elf section `usb_descriptor' will not fit in region `SRAM'

/opt/zephyr-sdk/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/8.3.0/../../../../arm-zephyr-eabi/bin/ld: section .intList VMA [0000000020006000,0000000020006047] overlaps section usb_descriptor VMA [0000000020005fcc,0000000020006099]

/opt/zephyr-sdk/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/8.3.0/../../../../arm-zephyr-eabi/bin/ld: region `SRAM' overflowed by 228 bytes

collect2: error: ld returned 1 exit status

%

zephyr/CMakeFiles/zephyr_prebuilt.dir/build.make:97: recipe for target 'zephyr/zephyr_prebuilt.elf' failed

make[2]: *** [zephyr/zephyr_prebuilt.elf] Error 1

CMakeFiles/Makefile2:662: recipe for target 'zephyr/CMakeFiles/zephyr_prebuilt.dir/all' failed

make[1]: *** [zephyr/CMakeFiles/zephyr_prebuilt.dir/all] Error 2

Makefile:83: recipe for target 'all' failed

make: *** [all] Error 2

script returned exit code 2

```

hakehuang reviewed (approved):
except one usb sample code issue, there is no other exist application issue on this porting.

jhqian committed:
soc: arm: nxp: k32l2: add soc support k32l2b

- Add device support for K32L2B

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

jhqian committed:
boards: arm: add board support frdm_k32l2b3

- Add board support for frdm_k32l2b3
- Add board specific pinmux, doc and dts.
- Tested with hello_world, synchronization, philosophers, basic/blinky
and basic/button.

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

jhqian commented:
Not sure what happened on the CI system side, the failed tickless_concept can't be reproduced on my side.

jhqian commented:
Not sure about the error reported in during shippable checking.
